openbb_sdk/providers/tests/test_provider_field_dupes.py

```diff
@@ -0,0 +1,189 @@
+            if provider_fields:
+                map_[cls.__name__].append({subclass.__name__: provider_fields})
+
+
+class ProviderFieldDupesTest(unittest.TestCase):
+    """Test for duplicate fields in the provider models."""
+
+    def test_provider_field_dupes(self):
+        """Check for duplicate fields in the provider models.
+
+        This function checks for duplicate fields in the provider models
+        and identifies the fields that should be standardized.
+        """
+        standard_models_directory = os.path.dirname(standard_models.__file__)
+        standard_models_files = glob.glob(
+            os.path.join(standard_models_directory, "*.py")
+        )
+
+        standard_query_classes = get_subclasses(
+            standard_models_files, standard_models.__name__, QueryParams
+        )
+        standard_data_classes = get_subclasses(
+            standard_models_files, standard_models.__name__, Data
+        )
+
+        provider_model_files = []
+        providers_path = providers.__path__[0]
+
+        # Walk through the directory tree
+        for root, _, files in os.walk(providers_path):
+            if "models" not in root:
+                continue
+
+            # Use fnmatch to find .py files
+            for filename in fnmatch.filter(files, "*.py"):
+                provider_model_files.append(os.path.join(root, filename))
+
+        child_parent_dict = {}
+
+        for file in provider_model_files:
+            splited_path = file.split("/")
+            mod = splited_path[-1].split(".")[0]
+            models_dir = splited_path[-2]
+            openbb_prov = splited_path[-3]
+            prov = splited_path[-4]
+
+            provider_module = importlib.import_module(
+                f"{providers.__name__}.{prov}.{openbb_prov}.{models_dir}.{mod}"
+            )
+
+            # query classes
+            child_parent_map(child_parent_dict, standard_query_classes, provider_module)
+
+            # data classes
+            child_parent_map(child_parent_dict, standard_data_classes, provider_module)
+
+        # remove keys with no values
+        child_parent_dict = {k: v for k, v in child_parent_dict.items() if v}
+
+        for std_cls in child_parent_dict:
+            with self.subTest(i=std_cls):
+                providers_w_fields = child_parent_dict[std_cls]
+
+                fields = []
+                provider_models = []
+
+                for provider_cls in providers_w_fields:
+                    provider_models.extend(list(provider_cls.keys()))
+                    fields.extend(list(provider_cls.values())[0])
+
+                seen = set()
+                dupes = [x for x in fields if x in seen or seen.add(x)]
+
+                assert not dupes, (
+                    f"The following fields are duplicated and should be standardized: {dupes}.\n"
+                    f"Standard model: {std_cls}, Provider models: {', '.join(provider_models)}\n"
```

Comment:
We could show the providers and the fields collectively that are found rather than just the fields!